test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,576 @@
+			},
+			Annotations: map[string]string{
+				"openshift.io/host.generated": "true",
+			},
+		},
+		Spec: routev1.RouteSpec{
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromString("proxy"),
+			},
+			TLS: &routev1.TLSConfig{
+				Termination: "edge",
+			},
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: "apicast-example-apicast",
+			},
+			Host: "selfmanaged-staging" + hostsub,
+		},
+	}
+	err := client.Create(ctx, route)
+	if err != nil {
+		log.Error("Error create Apicast Route", err)
+		return "", err
+	}
+	routeHost := route.Spec.Host
+	log.Info("routeHost: " + routeHost)
+	return routeHost, nil
+}
+
+func customerLogin() error {
+	log.Info("Customer Login")
+	token, err := getCustomerAminPasswordTokenFromTerm("Enter Customer Admin user Token, and press Enter: ")
+	if err != nil {
+		log.Error("Error login customer, can't get token: ", err)
+		return err
+	}
+	command := "oc login --token=" + token
+	err = run_shell_command(command)
+	if err != nil {
+		log.Error("Error login customer: ", err)
+		return err
+	}
+	return nil
+}
+
+func getCustomerAminPasswordTokenFromTerm(prompt string) (string, error) {
+	fmt.Print(prompt)
+	bytepw, err := term.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		return "", err
+	}
+	pass := string(bytepw)
+	return pass, nil
+}
+
+func run_shell_command(command string) error {
+	log.Info(command)
+	cmd := exec.Command("bash", "-c", command)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		log.Error("Error: ", err)
+		return err
+	}
+	return nil
+}
+
+// Configure the service to use the self-managed APIcast instead of the built-in APIcast for API.
+// See https://access.redhat.com/documentation/en-us/red_hat_openshift_api_management/1/topic/a702e803-bbc8-47af-91a4-e73befd3da00
+// step 11
+func promoteSelfManagedAPIcastInUI() (string, error) {
+	message := "This is the manual step - Configure the service to use the self-managed APIcast instead of the built-in APIcast for API. \n"
+	message += "a. Navigate to 3scale Admin Portal. You can use the following command to find route  \"oc get routes --namespace redhat-rhoam-3scale | grep admin\" \n"
+	message += "b. In the Products section, click API → Integration → Settings → APIcast Self Managed.\n"
+	message += "c. Change the Staging Public Base URL. Replace api-3scale-apicast- with selfmanaged-. \n"
+	message += "d. Click Update Product. \n"
+	message += "e. Click API → Configuration → Promote to Staging \n"
+	message += "f. Copy user_key value (from Staging APIcast - Example curl for testing) to command prompt \n"
+	log.Info(message)
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Print("Waiting for manual step completion. Copy user_key value here and Press enter when done :")
+	userKey, err := reader.ReadString('\n')
+	if err != nil {
+		log.Error("Error: ", err)
+		return "", err
+	}
+	if len(userKey) > 0 {
+		userKey = userKey[:len(userKey)-1]
+	}
+	log.Info("user_key: " + userKey)
```

Comment:
for command line output I think that passwords are usually masked. Not sure we should output it here?

Reply:
User key is application key and it will be printed also in Validation in route, like:
INFO[0230] HTTP Request: https://selfmanaged-staging.apps.valerym01.l80z.s1.devshift.org/?user_key=c166b93900e4893184e9e19c7b26b967 
INFO[0231] Response Code: 200                           
INFO[0231] Response Body: {
  "method": "GET",
  "path": "/",
  "args": "user_key=c166b93900e4893184e9e19c7b26b967",

@laurafitzgerald , @psturc  this is the script that is running by User, and User is taking this key from 3scale Portal. In interactive mode , maybe we don't need to care about security, Or we need suppress also HTTP response, as it's printing user_key.  If we want to do it now - we will replace userKey, err := reader.ReadString('\n') (line 489), with userKey, err := term.ReadPassword(int(syscall.Stdin)) and return string(userKey[:]), err. 
I can do all this, but need decide if this secure approach is required  in interactive mode.

Reply:
Done. No printing of any secure key key/token/passwords